File: app/src/main/java/com/przemolab/oknotifier/NotifierApp.java
package com.przemolab.oknotifier;

import android.app.Application;

import com.przemolab.oknotifier.modules.NotifierRepositoryModule;
import com.przemolab.oknotifier.modules.OpenKattisServiceModule;

import timber.log.Timber;

public class NotifierApp extends Application {

    public AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        appComponent = DaggerAppComponent.builder()
                .openKattisServiceModule(new OpenKattisServiceModule())
                .notifierRepositoryModule(new NotifierRepositoryModule(this))
                .build();
        appComponent.inject(this);

        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }
    }
}

File: app/src/main/java/com/przemolab/oknotifier/NotifierApp.kt
package com.przemolab.oknotifier

import android.app.Application

import com.przemolab.oknotifier.modules.NotifierRepositoryModule
import com.przemolab.oknotifier.modules.OpenKattisServiceModule

import timber.log.Timber

class NotifierApp : Application() {

    lateinit var appComponent: AppComponent

    override fun onCreate() {
        super.onCreate()

        appComponent = DaggerAppComponent.builder()
                .openKattisServiceModule(OpenKattisServiceModule())
                .notifierRepositoryModule(NotifierRepositoryModule(this))
                .build()
        appComponent.inject(this)

        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }
    }
}
